docs/regression_test_api.rst

```diff
@@ -36,71 +36,97 @@ Pipeline Hooks
-   Instead, if no parameter with a matching name exists in any of the parent parameter spaces, a new regression test parameter is created.
+  Inserts a new regression test variable.
 
-   :param values: A list containing the parameter values.
-       If no values are passed when creating a new parameter, the parameter is considered as *declared* but not *defined* (i.e. an abstract parameter).
-       Instead, for an existing parameter, this depends on the parameter's inheritance behaviour and on whether any values where provided in any of the parent parameter spaces.
-   :param inherit_params: If :obj:`False`, no parameter values that may have been defined in any of the parent parameter spaces will be inherited.
-   :param filter_params: Function to filter/modify the inherited parameter values that may have been provided in any of the parent parameter spaces.
-       This function must accept a single argument, which will be passed as an iterable containing the inherited parameter values.
-       This only has an effect if used with ``inherit_params=True``.
+  .. code:: python
 
+    class Foo(rfm.RegressionTest):
+        my_var = variable(int, value = 8)
+
+        def __init__(self):
+            print(self.my_var) # prints 8.
 
-.. py:class:: reframe.core.pipeline.RegressionTest.variable(*types, value=None, field=None)
+  The argument ``value`` sets the default value for the variable.
+  A variable may not be declared more than once. However, it is possible to alter a variable's value after it was declared by simply assigning it a new value directly in the class body.
 
-   Inserts a new regression test variable.
-   The argument ``value`` sets the default value for the variable.
-   A variable may not be declared more than once. However, it is possible to alter a variable's value after it was declared by simply assigning it a new value directly in the class body.
-   A variable may be set as required by simply assigning a variable as `required`.
-   Note that a variable must be defined before is referenced in the regression test. Otherwise, an :py:exc:`AttributeError` will be raised.
+  .. code:: python
 
-   :param types: the supported types for the variable.
-   :param value: the default value assigned to the variable.
-   :param field: the field validator to be used for this variable.
-        If no field argument is provided, it defaults to
-        :class:`reframe.core.fields.TypedField`.
-        Note that the field validator provided by this argument must derive from
-        :class:`reframe.core.fields.Field`.
+    class Bar(Foo):
+        my_var = 4
+
+        def __init__(self):
+            print(self.my_var) # prints 4.
+
+  A variable may be set as required by simply assigning a variable as `required`.
+  Note that a variable must be defined before is referenced in the regression test. Otherwise, an :py:exc:`AttributeError` will be raised.
```

Comment:
This can be a bit confusing and have users declaring every variable that they would normally define with a `self.my_var` in their test. What I find missing in this section of the documentation is when somebody should use the `variable()` built-in and what does it offer?

Reply:
Agree. I've added quite a bit more of text in this section.